### PR TITLE
Add ephemeral local storage resource name first

### PR DESCRIPTION
**Pull request by NickrenREN**
Since finally decided the ephemeral local storage resource name, send this PR to add the resource name first, and then i will rebase all my PRs about local storage isolation

**Special notes for your reviewer**:
related to: #50818 

**Release note**:
```release-note
NONE
```

/assign @jingxu97

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -3170,9 +3170,11 @@ const (
 	// Local Storage for scratch space, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
 	// The resource name for ResourceStorageScratch is alpha and it can change across releases.
 	ResourceStorageScratch ResourceName = "storage.kubernetes.io/scratch"
+	// Local Storage for ephemeral local storage space, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
```
Comment (jingxu97) [inline pkg/api/types.go]: I think just put Local ephemeral storage, in bytes. is fine.

Comment (jingxu97) [inline pkg/api/types.go]: also mention the name is alpha

File: pkg/api/types.go [hunk 2]
```diff
@@ -3722,10 +3724,14 @@ const (
 	ResourceRequestsMemory ResourceName = "requests.memory"
 	// Storage request, in bytes
 	ResourceRequestsStorage ResourceName = "requests.storage"
+	// ephemeral local storage request, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
```
Comment (jingxu97) [inline pkg/api/types.go]: local ephemeral storage?

**Comment by jingxu97** [comment]
lgtm

**Comment by NickrenREN** [comment]
ping @smarterclayton  @thockin  for approval

File: pkg/api/types.go [hunk 3]
```diff
@@ -3170,9 +3170,11 @@ const (
 	// Local Storage for scratch space, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
 	// The resource name for ResourceStorageScratch is alpha and it can change across releases.
 	ResourceStorageScratch ResourceName = "storage.kubernetes.io/scratch"
```
Comment (vishh) [inline pkg/api/types.go]: These resources need to be removed. @NickrenREN @jingxu97 can either you clean that up?

Comment (NickrenREN) [inline pkg/api/types.go]: Yes, i will send a separate PR to do that

**Pull request merged**